app/controllers/concerns/puppet/hosts_controller_extensions.rb

```diff
@@ -0,0 +1,182 @@
+module ForemanPuppetEnc
+  module Concerns::Puppet::HostsControllerExtensions
```

Comment:
This already exists as
```suggestion
  module Extensions::HostsControllerExtensions
```